### PR TITLE
IPDatabase.py: don't checkout specified version if current version newer

### PR DESCRIPTION
**Pull request by vogelpi**
This commit modifies `update_ips()` to check the timestamp of the
current IP version and the version specified in the manifest. If
the current IP version is newer than the specified version, the
checkout is not performed and a warning is printed.

This is useful if experimenting with newer IP versions in hierarchical
flows such as PULPIssimo. Without this commit, the user has to first
update and commit & push `ips_list.yml` of the higher-level IP, and
then do the same with the top-level `ips_list.yml` which is a lot of
overhead. Otherwise, the tool refuses to work with a newer IP version
and always checks out the older version specified in the `ips_list.yml`
of the higher-level IP.

### CODE REVIEW

**Comment by bluewww** [comment]
I agree with this change in principle. Still I think there should be a command line parameter called `--force` or something to get the old behaviour.

**Comment by vogelpi** [comment]
Hi @bluewww , that's a good idea. I will add a new parameter for this.

**Comment by vogelpi** [comment]
Hi @bluewww , I made the feature controllable by a new parameter `force_downgrade`.

File: ipstools/IPDatabase.py [hunk 1]
```diff
@@ -470,12 +470,15 @@ def remove_ips(self, skip_check=False):
         except OSError:
             print(tcolors.WARNING + "WARNING: Not removing %s as there are unknown IPs there." % (self.ips_dir) + tcolors.ENDC)
 
-    def update_ips(self, origin='origin'):
+    def update_ips(self, origin='origin', force_downgrade=False):
```
Comment (bluewww) [inline ipstools/IPDatabase.py]: Default should be `force_downgrade=True` in discussion with @meggiman

**Review by bluewww — CHANGES_REQUESTED** [review]

File: ipstools/IPDatabase.py (hunk 1 above)
Comment (FrancescoConti) [reply ipstools/IPDatabase.py]: Yes I entirely agree, in this way we keep the original behavior as default...

Comment (vogelpi) [reply ipstools/IPDatabase.py]: I know, it's a long time ago. I changed the parameter to True to get the same default behavior as before.

**Review by bluewww — APPROVED** [review]
Tested in on the PULP repository. LGTM.

**Pull request merged**